lua/neocodeium/commands.lua

```diff
@@ -182,6 +183,75 @@ end
 function M.toggle()
   options.enabled = not options.enabled
 end
+
+local function launch_chat(response)
+  local metadata = server:request_metadata()
+  local processes = vim.json.decode(response.out[1])
+  local chat_port = processes["chatClientPort"]
+  local ws_port = processes["chatWebServerPort"]
+
+  -- possible, server is not ready
+  if not (chat_port and ws_port) then
+    return
+  end
+
+  local server_opts = options.server
+  local has_enterprise_extension = (server_opts.api_url and server_opts.api_url ~= "") and true
+    or false
+
+  local url = vim
+    .iter({
+      api_key = metadata.api_key,
+      ide_name = metadata.ide_name,
+      ide_version = metadata.ide_version,
+      extension_name = metadata.extension_name,
+      extension_version = metadata.extension_version,
+      web_server_url = "ws://127.0.0.1:" .. ws_port,
+      has_enterprise_extension = has_enterprise_extension,
+      locale = "en",
+      ide_telemetry_enabled = true,
+      has_index_service = true,
+      app_name = "Neovim",
+      spen_file_pointer_enabled = true,
+      diff_view_enabled = true,
+    })
+    :fold("http://127.0.0.1:" .. chat_port .. "/?", function(acc, key, value)
+      return acc .. key .. "=" .. tostring(value) .. "&"
+    end)
+
+  vim.schedule(function()
+    open_browser(url)
```

Comment:
Just use `vim.ui.open()` here instead of vim.schedule and open_browser. I should remove completely this open_browser() function because it does more than intended and misleading.